sfcasts/ep1/github-service-part-1.md

```diff
@@ -1,62 +1,67 @@
-
-We can go back to our terminal and let's run the test one more time. Make sure that
-yes, instead of two failures, we only have one. Our GitHub service test test GitHub
-test get health report returns, correct health status for Dino with data set sick
-Dino, it failed to asserting that two variables reference the same object. And it's
-of course doing that because we are only returning a healthy status for get health
-reports. Now that we have this kind of wired up, let's go ahead and implement the
-service.
+# GitHub Service - Part 1
 
+Now that we can see if a `Dinosaur` is accepting visitors based on its
+current health. We need to keep our dino's health updated automatically using the
+health status labels that GenLab has applied to several issues on GitHub. To do 
+that we'll create a service that will grab those labels from the issues using
+GitHub's API.
+
+## Test for our service first
+
+To test our new service, add a `Service/` directory inside of `tests/Unit/` 
+and then create a new `GithubServiceTest` which will extend `TestCase`. Add method
+`testGetHealthReportReturnsCorrectHealthStatusForDino` and inside, even
+though it doesn't exist yet, `$service = new GithubService()`.
+
+Our service will return a `HealthStatus` enum that is created from the health status
+label on GitHub, so we'll `assertSame()` that `$expectedStatus` is identical to 
+`$service->getHealthReport()`. Yes, we'll be using a data provider for this test...
+And because our service will return a *single* `HealthStatus` object, let's tell 
+the method which dino we need a health report for by passing in `$dinoName` to
+`getHealthReport()`.
+
+Create a `dinoNameProvider()` which returns a `\Generator` and our first dataset
+will be a `Sick Dino` with `HealthStatus::SICK`, use `Daisy` for the dino's name.
+Next up is a `Healthy Dino` with `HealthStatus::HEALTHY` who happens to be the
+one and only `Maverick`. Now add the `@dataProvider` annotation so our test uses 
+the `dinoNameProvider` and we'll pass in a `HealthStatus $expectedStatus`
+argument and a `string` called `$dinoName`.
+
+Move to your terminal and run the test:
+
+```terminal
+./vendor/bin/phpunit
+```
+
+And... Yup - we have two errors because: 
+
+> The GithubService class cannot be found
+
+## Create the service that will call GitHub
+
+Let's fix that by creating a new `Service/` directory inside of `src/` and then 
+we'll need a new `GithubService` class. Inside, add a `getHealthReport()` method
+which takes a `string $dinosaurName` and gives back a `HealthStatus` object.
+
+Hmm... Alrighty, the first thing we will do is call GitHub's API to get the list
+of issues for the `dino-park` repository, then filter those issues to pick the 
+one that matches `$dinosaurName` and last but not least... return 
+`GithubStatus::HEALTHY`, unless we find GenLab has added a `Status: Sick` label
+to an issue that matches our dinos name.
+
```

Comment:
We aren't actually coding here in the video. We are however, mapping out (using `// coments`) what we will code in the next chapter